### ISSUE TITLE
Clarify extended-use standard structures

### ISSUE THREAD

**Issue by tychonievich** [FamilySearch/GEDCOM#13]
The current specification defines

> An **extended-use standard structure** is a structure whose type, tag, and meaning are defined in this document and whose superstructure is a tagged extension structure.

and recommends

> It is recommended that extensions use extended-use standard structures instead of tagged extension structures if extended-use standard structures will suffice.

However, it does not clarify how to disambiguate these extended-use substructures. For example, should `_XYZ`.`DATE` be interpreted as a `g7:DATE`, `g7:DATE-exact`, `g7:HEAD-DATE`, `g7:NO-DATE`, or `g7:DATA-EVEN-DATE`?

Comparing the current specification with the last pre-URI draft, I believe the following is the intended meaning:

- The URI of a standard tag is determined by the combination of (a) the specification, (b) the superstructure's structure type, and (c) it's tag.
- If the superstructure's structure type is one defined in the specification then, regardless of its tag, it's mapping between substructure tags and substructure structure types is given in the specification.
- Otherwise, if there is a URI defined in the spec as `g7:`*tag*, that is the URI of the structure type of this extended-use standard structure.

For example, that rule would interpret the following example:

```gedcom
0 @1@ _LOC
1 NAME New Amsterdam
2 DATE FROM 1609 TO 1664
1 TYPE Trading post
```

as the following structure hierarchy:

- a tagged extension structure
    - a `g7:NAME`
        - an error; `g7:NAME` has no `DATE`-tagged substructures
    - a `g7:TYPE`

and the following example

```gedcom
0 HEAD
1 GEDC
2 VERS 7.0
1 SCHMA
2 TAG _CHAN https://gedcom.io/terms/v7/CHAN
0 @I1@ INDI
1 NAME Mary //
2 _CHAN
3 DATE JUN 2021
```

as an error because the `DATE` that is a substructure of `g7:CHAN` is `g7:DATE-exact` and the given date lacks a day of the month.

Is this the behavior we believe the 7.0 specification implies? If not, how should it be adjusted? If it is, we'll need to consider how to make this clear in the specification itself.

**Comment by tychonievich** [comment]
Discussed 2021-06-22
Alternative views:
- When you use a standard tag under an extension tag, it must match the semantic meaning and payload rules of at least one standard structure type with that tag, but may have distinct substructures and other limitations imposed by the extension.
or
- This reveals an inconsistency in the "extended-use standard structure" concept, so it should be removed
or
- Once you hit an extension tag, all of its substructures are also extensions even if they use standard tags

We did not come to a conclusion which the right view is.

**Comment by tychonievich** [comment]
Proposals on how to change "An **extended-use standard structure** is a structure whose type, tag, and meaning are defined in this document and whose superstructure is a tagged extension structure."

1. Change it to "An **extended-use standard structure** is a structure whose type, tag, and meaning are defined in this document and whose superstructure is a tagged extension structure <ins>or an extended-use standard structure</ins>."; also clarify how the URIs of these structures are selected 
2. Change it to "<ins>A **standard-tagged extension structure** is a structure with a standard tag whose superstructure is either a tagged extension structure or an extended-use standard structure</ins>."; also add a modification to SCHMA to allow defining the URIs of these structures (e.g. by letting dot-separated tags in the TAG structure or letting a TAG structure have TAG substructures, such as
    ````gedcom
    1 SCHMA
    2 TAG _LOC <url _LOC>
    2 TAG _LOC.NAME <url of _LOC.NAME>
    ````
    or
    ````gedcom
    1 SCHMA
    2 TAG _LOC <url _LOC>
    3 TAG NAME <url of _LOC.NAME>
    ````
    I'm not sure which I like better)
3. Change the text as per 2, but don't provide any way to get a URI for these and instead deprecate this usage.

**Comment by tychonievich** [comment]
Discussed 2021-06-29
Two part solution:

1. Clarify how you get the structure type (see the pull request)
2. Adjust extended-use as follows:

    An **extended-use standard structure** is a structure whose type, tag, and meaning are defined in this document and whose superstructure is a tagged extension structure or another extended-use standard structure.
    
    :::note
    In practice, this means that if a structure is underneath an extnesion structure, regardless of how many levels up, then the structure must follow all rules of the structure type *except* it may have any types of substructures.
    :::

**Comment by tychonievich** [comment]
Following up:
I think this proposal and the follow-on conversation we had in the 2021-06-29 meeting means there are five kinds of structure types:

1. Standard, as defined in the spec
2. Extension-tagged standard, using a documented extension tag to get a URI in the spec. They can appear as substructures of structures that cannot accept the structure type with a standard tag, but abide all other standard restrictions.
3. Undocumented extension. Anything goes.
4. Documented extension, using a documented extension tag to get a URI _not_ in the spec. Constrained by an external spec.
5. Extended-use standard, using a standard tag and appearing somewhere beneath an extension. They can have a superstructure and/or substructures of types not documented for their structure type in the spec, but abide all other standard restrictions.

Extended-use standard use a structure type derived as
- the URI for that tag as a substructure of its superstructure's type somewhere in the spec; or, if that doesn't exist
- the URI crated by concatenating `https://gedcom.io/terms/v7/` and the tag if that is defined as a structure type in the spec; or, if that doesn't exist
- it's an error

Cardinality rule `{1:` requires a standard tag (standard structure or extended-use standard structure; using only an extension-tagged standard structure does not suffice).
Cardinality rule `:1}` limits a structure type (the sum of the number of standard, extension-tagged standard, and extended-use standard structures of that type cannot exceed 1).

I was going to put this directly in a pull request, but realized it needs a bit of reordering between sections 1.5 and 1.5.1, and possibly a full re-write of section 1.5, which will take time. I wanted to post the objective of that pull request before I invest that effort in writing it.

**Comment by tychonievich** [comment]
We now have two general approaches to clarifying extended-use standard structures (EUSS):

PR #19 provides detailed rules for EUSS:
- Explains how to pick a URI
- Explains substructure, superstructure, and cardinality rules
- Result: I can edit substructures of extensions even if I don't understand the extension
- Result: `_LOC`.`NAME`.`DATE` is OK, and is a `g7:DATE`
- Result: `_LOC`.`TYPE` is OK, and is a `g7:TYPE` -- which is not what GEDCOM-L uses `_LOC`.`TYPE` for.
- Also: explains rules about enumerations and relocated standard structures

PR #24 removes EUSS entirely
- Once you hit an extension tag, all substructures are defined by the spec for that tag, not by the v7 spec
- Result: I can only edit substructures of extensions if I understand the extension
- Result: `_LOC`.`NAME`.`DATE` and `_LOC`.`TYPE` are both OK, and both are defined by the `_LOC` spec
- Also: explains rules about relocated standard structures

**Comment by tychonievich** [comment]
Resolved by #24
